modules/core/src/main/java/org/pixel/graphics/render/SpriteBatch.java

```diff
@@ -157,33 +196,31 @@ private void flush() {
         }
 
         // draw the sprite data..
-        for (int i = 0, count = 0; i < bufferCount; i++) {
+        for (int i = 0, count = 0; i < bufferCount; ++i) {
             SpriteData spriteData = this.spriteData[i];
             if (!spriteData.active) {
                 continue; // skip inactive sprites
             }
 
-            // texture checkup:
             if (lastTextureId != spriteData.textureId) {
-                // is there any previous texture id?
-                if (lastTextureId >= 0) {
-                    // yes, which means we have something to render..
-                    flushBatch(count);
-                    count = 0;
-                    dataBuffer.clear();
+                if (!shaderTextureMap.containsKey(spriteData.textureId)) {
+                    if (shaderTextureMap.size() >= shaderTextureCount) {
+                        flushBatch(count);
+                        count = 0;
+                        dataBuffer.clear();
+                        shaderTextureMap.clear();
```

Comment:
Can't previous textures be re-used later? Example: max 2 textures, 3 different textures used repeatedly, meaning all but the third texture is cleared from the map and all other textures need to be loaded, as opposed to just reloading once every third texture.

Reply:
We could implement a "capacity" hash map were old entries are removed (instead of clearing all when this condition happens). But in my opinion, is mostly a user design problem as textures would still be swapping (on that scenario 1 by 1).